### DISCUSSION TITLE
Try out 2.0.0!

### DISCUSSION THREAD

**Discussion by nipafx** [junit-pioneer/junit-pioneer#701] [General]
Hey folks,

we recently published [the first release candidate for version 2.0](https://github.com/junit-pioneer/junit-pioneer/releases/tag/v2.0.0-RC1) and call on Pioneer users to **try it out and give us feedback**. These are the most noteworthy changes over the last 1.x release (1.9.1):

* required Java version is now 11
* the module descriptor (`module-info.class`) is now in the JAR's root directory
* the [old variant](https://junit-pioneer.org/docs/cartesian-product-v1/) of the Cartesian test extension was deprecated in 1.6.0 and is now removed - users must migrate to the [new variant](https://junit-pioneer.org/docs/cartesian-product/)
* the [default locale extension](https://junit-pioneer.org/docs/default-locale-timezone/#defaultlocale) no longer uses the deprecated `Locale` constructors but the more stringent `Locale.Builder`, which means tests may fail during setup - update their locale string to match one that `Locale.Builder` would accept

If these (or any other changes) cause you unexpected problems, please let us know. Also, if you try it out and everything works, that is also an important signal, so leave a comment in that case, too.

**Comment by nipafx** [comment]
I got the first (potential) issue. [I updated the demo project to RC1](https://github.com/junit-pioneer/junit-pioneer-demo/commit/1c42646150613acdd197737b37ce47601db1c419) and the Jackson capability thingy doesn't work for me:

```
Execution failed for task ':test'.
> Could not resolve all files for configuration ':testRuntimeClasspath'.
   > Could not resolve org.junit-pioneer:junit-pioneer.
     Required by:
         project :
      > Unable to find a variant of org.junit-pioneer:junit-pioneer:2.0.0-RC1 providing the requested capability org.junit-pioneer:junit-pioneer-jackson:
           - Variant compile provides org.junit-pioneer:junit-pioneer:2.0.0-RC1
           - Variant runtime provides org.junit-pioneer:junit-pioneer:2.0.0-RC1
           - Variant platform-compile provides org.junit-pioneer:junit-pioneer-derived-platform:2.0.0-RC1
           - Variant platform-runtime provides org.junit-pioneer:junit-pioneer-derived-platform:2.0.0-RC1
           - Variant enforced-platform-compile provides org.junit-pioneer:junit-pioneer-derived-enforced-platform:2.0.0-RC1
           - Variant enforced-platform-runtime provides org.junit-pioneer:junit-pioneer-derived-enforced-platform:2.0.0-RC1
```

Interestingly, it doesn't work for 1.9.0 either but for 1.9.1. 🤔

**Reply by beatngu13** [reply]
I have added a minimal GitHub Actions build for Maven and Gradle (see https://github.com/junit-pioneer/junit-pioneer-demo/commit/13e9b6f2c5529bb7f34c3d418202571ced92fe2b). Both passed, but locally Gradle was failing for me, too. Looks like `mavenLocal()` was a problem (see https://github.com/junit-pioneer/junit-pioneer-demo/commit/1e0bf431d403559a40045f2774908ca288f40438), now it also builds locally. 🎉

**Reply by nipafx** [reply]
Oh neat! Thank you for looking into this! ❤️

**Comment by sormuras** [comment]
> - the module descriptor (module-info.class) is now in the JAR's root directory

Modular usage is (still/now) working like a charm.

https://github.com/sormuras/bach/commit/8db9237f847310ebbf5e98aedd0eb32abd97f8e3

**Reply by nipafx** [reply]
Thanks for letting us know! ❤️
